paasta_tools/spark_tools.py

```diff
@@ -247,3 +249,24 @@ def get_spark_driver_monitoring_labels(
         "spark.yelp.com/driver_ui_port": ui_port_str,
     }
     return labels
+
+
+def get_spark_memory_in_unit(mem: str, unit: Literal["k", "m", "g", "t"]) -> float:
+    """
+    Converts Spark memory to the desired unit.
+    mem is the same format as JVM memory strings: just number or number followed by 'k', 'm', 'g' or 't'.
+    unit can be 'k', 'm', 'g' or 't'.
+    Returns memory as a float converted to the desired unit.
+    """
+    memory_bytes = 0.0
+    if mem:
+        if mem[-1] in MEM_MULTIPLIER:
+            memory_bytes = float(mem[:-1]) * MEM_MULTIPLIER[mem[-1]]
```

Comment:
if we don't add validation for the  user-provided value here now, then this needs to be wrapped in a try-except as well (since otherwise this will crash if a user puts in "lolk" or `1B` as a `spark.driver.memory` value)